### PR TITLE
1. check celery status before running tasks 2. random->lhs when lacking data for GPR 3. fix editing session

### PR DESCRIPTION
**Pull request by yangdsh**
### 1. check celery status before running tasks
If celery is not running, first I used `call_command()` to stop celery and celerybeat.
I did not use `call_command()` to restart celery, because `call_command()` is a blocking function call and it blocks Django from running the following lines. Instead, I run celery in the background using `os.popen('python manage.py startcelery &')`

### 2. random->lhs when lacking data for this workload
If there is no background task for this workload, we generate 10 recommendation configs with the lhs algorithm instead of the random algorithm. 

### 3. query and edit the session hyperparameters
I feel that we can use the existing [backdoor method](https://github.com/cmu-db/ottertune/blob/06948089d2aaec09aca37eb7e7c59c6e53ea0fae/client/driver/fabfile.py#L732)  in the driver fabfile to query and edit the session hyperparameters. The arguments format should be as follows:
1. There are ' around the json string
2. There is \ before any comma in the json string
3. There is no \ before " in the json string

***Examples:***
- querying command:
fab edit_website_session:username=user,password=abcd123,project_name=oracle,dbms_type=Oracle,dbms_version=12.2.0.1.0,name=session_name,verbose=True
- editing command: 
fab edit_website_session:username=user,password=abcd123,project_name=oracle,dbms_type=Oracle,dbms_version=12.2.0.1.0,name=session_name,hyperparameters='{    "DDPG_ACTOR_HIDDEN_SIZES": [128\, 128\, 64]\,    "DDPG_ACTOR_LEARNING_RATE": 0.02\,    "DDPG_CRITIC_HIDDEN_SIZES": [64\, 128\, 64]\,    "DDPG_CRITIC_LEARNING_RATE": 0.001\,    "DDPG_BATCH_SIZE": 32\,    "DDPG_GAMMA": 0.0\,    "DDPG_SIMPLE_REWARD": true\,    "DDPG_UPDATE_EPOCHS": 30\,    "DDPG_USE_DEFAULT": false\,    "DNN_DEBUG": true\,    "DNN_DEBUG_INTERVAL": 100\,    "DNN_EXPLORE": false\,    "DNN_EXPLORE_ITER": 500\,    "DNN_GD_ITER": 100\,    "DNN_NOISE_SCALE_BEGIN": 0.1\,    "DNN_NOISE_SCALE_END": 0.0\,    "DNN_TRAIN_ITER": 100\,    "FLIP_PROB_DECAY": 0.5\,    "GPR_BATCH_SIZE": 3000\,    "GPR_DEBUG": true\,    "GPR_EPS": 0.001\,    "GPR_EPSILON": 1e-06\,    "GPR_LEARNING_RATE": 0.01\,    "GPR_LENGTH_SCALE": 2.0\,    "GPR_MAGNITUDE": 1.0\,    "GPR_MAX_ITER": 500\,    "GPR_MAX_TRAIN_SIZE": 7000\,    "GPR_MU_MULTIPLIER": 1.0\,    "GPR_MODEL_NAME": "BasicGP"\,    "GPR_HP_LEARNING_RATE": 0.001\,    "GPR_HP_MAX_ITER": 5000\,    "GPR_RIDGE": 1.0\,    "GPR_SIGMA_MULTIPLIER": 1.0\,    "GPR_UCB_SCALE": 0.2\,    "GPR_USE_GPFLOW": true\,    "GPR_UCB_BETA": "get_beta_td"\,    "IMPORTANT_KNOB_NUMBER": 10000\,    "INIT_FLIP_PROB": 0.3\,    "NUM_SAMPLES": 30\,    "TF_NUM_THREADS": 4\,    "TOP_NUM_CONFIG": 10}'

### CODE REVIEW

File: server/website/website/utils.py [hunk 1]
```diff
@@ -494,3 +495,23 @@ def model_to_dict2(m, exclude=None):
...
+    if 'OK' in celery_status:
+        return 'celery is running'
+
+    retries = 0
+    while retries < 5:
+        LOG.warning('Celery is not running.')
+        retries += 1
+        call_command('stopcelery')
+        os.popen('python manage.py startcelery &')
+        time.sleep(5 * retries)
```
Comment (bohanjason) [inline server/website/website/utils.py]: I try to restart celery on my local machine, it takes about 30 seconds to finish restarting. 
I think we can set the interval to 30s ?   
<code> time.sleep(30)  </code>

File: server/website/website/utils.py [hunk 2]
```diff
@@ -494,3 +495,23 @@ def model_to_dict2(m, exclude=None):
         if fname not in exclude:
             d[fname] = getattr(m, fname, None)
     return d
+
+
+def check_and_run_celery():
+    celery_status = os.popen('python manage.py celery inspect ping').read()
```
Comment (bohanjason) [inline server/website/website/utils.py]: we may use python3 instead of python ?

File: server/website/website/utils.py (hunk 1 above)
Comment (yangdsh) [reply server/website/website/utils.py]: Thanks!

File: client/driver/fabfile.py [hunk 3]
```diff
@@ -663,6 +663,10 @@ def _http_content_to_json(content):
     return json_content, decoded
 
 
+def _parse_bool(val):
```
Comment (dvanaken) [inline client/driver/fabfile.py]: This function is now located in utils.py: https://github.com/cmu-db/ottertune/blob/master/client/driver/utils.py#L50

File: server/website/website/tasks/async_tasks.py [hunk 4]
```diff
@@ -190,11 +190,14 @@ def aggregate_target_results(result_id, algorithm):
...
-    if newest_result.session.tuning_session == 'lhs':
+    if not has_pipeline_data or newest_result.session.tuning_session == 'lhs':
+        if not has_pipeline_data and newest_result.session.tuning_session == 'tuning_session':
+            LOG.debug("Background tasks haven't ran for this workload yet, picking random data.")
+
         all_samples = JSONUtil.loads(newest_result.session.lhs_samples)
         if len(all_samples) == 0:
             knobs = SessionKnob.objects.get_knobs_for_session(newest_result.session)
-            all_samples = gen_lhs_samples(knobs, 100)
+            all_samples = gen_lhs_samples(knobs, 10)
```
Comment (dvanaken) [inline server/website/website/tasks/async_tasks.py]: We should generate 100 samples if `newest_result.session.tuning_session == 'lhs'` or 10 samples if `not has_pipeline_data`

**Comment by dvanaken** [comment]
@yangdsh I accidentally hit 'approved' review but please make the couple of changes I requested and I will merge. Thanks again!

**Pull request merged**